heracles/dices/jackknife.py

```diff
@@ -359,12 +350,16 @@ def _debias_covariance(cov_jk, Q):
     Internal method to debias the Jackknife covariance.
     Useful when the delete2 correction is already computed.
     """
-    debiased_cov = {}
-    for key in list(cov_jk.keys()):
-        c = cov_jk[key].array - Q[key].array
-        debiased_cov[key] = Result(
-            c,
-            ell=cov_jk[key].ell,
-            axis=cov_jk[key].axis,
-        )
+    debiased_cov = deepcopy(cov_jk)
+    for key in Q:
+        q = Q[key].array
+        c = debiased_cov[key].array
+        *_, length = q.shape
+        q_diag = np.diagonal(q, axis1=-2, axis2=-1)
+        c_diag = np.diagonal(c, axis1=-2, axis2=-1)
+        # Indices for the diagonal
+        diag_indices = np.arange(length)
+        # abs is only needed when too few Jackknife regions are used
+        c[..., diag_indices, diag_indices] = abs(c_diag - q_diag)
```

Comment:
Isn't this materially changing the output of the debiasing step? In the sense that the absolute value is something we are making up on the spot to fix a "problem" that is really intrinsic to the process?

Reply:
for large enough jk regions, q_diag should always be smaller than c_diag. The goal here is to cover for when only a few regions are used. We could something more kosher such as enfocing pos-def for each entry in the matrix stack but this would require more involved coding.